tests/t0123_gateway_json_cbor_test.go

```diff
@@ -788,73 +790,147 @@ func TestNativeDag(t *testing.T) {
-			  curl -sX GET "http://127.0.0.1:$GWAY_PORT/ipfs/$CID?format=dag-$format" -o ipfs_output &&
-			  curl -sX GET "http://127.0.0.1:$GWAY_PORT/ipns/$IPNS_ID?format=dag-$format" -o ipns_output &&
-			  test_cmp ipfs_output ipns_output
-			  '
-			*/
-			/**
-			  test_expect_success "GET $name from /ipns with explicit application/vnd.ipld.dag-$format has expected headers" '
-			  curl -svX GET -H "Accept: application/vnd.ipld.dag-$format" "http://127.0.0.1:$GWAY_PORT/ipns/$IPNS_ID" >/dev/null 2>curl_output &&
-			  test_should_not_contain "Cache-Control" curl_output &&
-			  test_should_contain "< Content-Type: application/vnd.ipld.dag-$format" curl_output &&
-			  test_should_contain "< Etag: \"${CID}.dag-$format\"" curl_output &&
-			  test_should_contain "< X-Ipfs-Path" curl_output &&
-			  test_should_contain "< X-Ipfs-Roots" curl_output
-			  '
-			*/
-			// TODO: IPNS
-			/**
-			  # When Accept header includes text/html and no explicit format is requested for DAG-(CBOR|JSON)
-			  # The gateway returns generated HTML index (see dag-index-html) for web browsers (similar to dir-index-html returned for unixfs dirs)
-			  # As this is generated, we don't return immutable Cache-Control, even on /ipfs (same as for dir-index-html)
-
-			  test_expect_success "GET $name on /ipfs with Accept: text/html returns HTML (dag-index-html)" '
-			  curl -sD - -H "Accept: text/html" "http://127.0.0.1:$GWAY_PORT/ipfs/$CID" > curl_output 2>&1 &&
-			  test_should_not_contain "Content-Disposition" curl_output &&
-			  test_should_not_contain "Cache-Control" curl_output &&
-			  test_should_contain "Etag: \"DagIndex-" curl_output &&
-			  test_should_contain "Content-Type: text/html" curl_output &&
-			  test_should_contain "</html>" curl_output
-			  '
-			*/
-			// TODO: IPNS
-			/**
-			  test_expect_success "GET $name on /ipns with Accept: text/html returns HTML (dag-index-html)" '
-			  curl -sD - -H "Accept: text/html" "http://127.0.0.1:$GWAY_PORT/ipns/$IPNS_ID" > curl_output 2>&1 &&
-			  test_should_not_contain "Content-Disposition" curl_output &&
-			  test_should_not_contain "Cache-Control" curl_output &&
-			  test_should_contain "Etag: \"DagIndex-" curl_output &&
-			  test_should_contain "Content-Type: text/html" curl_output &&
-			  test_should_contain "</html>" curl_output
-			  '
-			*/
+			// test_expect_success "GET $name on /ipfs with Accept: text/html returns HTML (dag-index-html)" '
+			// curl -sD - -H "Accept: text/html" "http://127.0.0.1:$GWAY_PORT/ipfs/$CID" > curl_output 2>&1 &&
+			// test_should_not_contain "Content-Disposition" curl_output &&
+			// test_should_not_contain "Cache-Control" curl_output &&
+			// test_should_contain "Etag: \"DagIndex-" curl_output &&
+			// test_should_contain "Content-Type: text/html" curl_output &&
+			// test_should_contain "</html>" curl_output
+			// '
+			{
+				Name: Fmt("GET {{name}} on /ipfs with Accept: text/html returns HTML (dag-index-html)", row.Name),
```

Comment:
Note: this is a bug and it should actually redirect. See https://github.com/ipfs/boxo/pull/315#pullrequestreview-1451557123.

Reply:
@hacdias I used a trailing slash, this is just for this one case, "Accept: text/html returns HTML (dag-index-html)",
correct?

Reply:
Yup